.github/workflows/ci.yml

```diff
@@ -110,3 +112,39 @@ jobs:
         timeout_minutes: 15
         max_attempts: 2
         command: ./tools/test_monitor/run-tests.sh
+  flow-legacy-client-compatibility-test:
+    name: Client Compatibility Tests With Observer
+    strategy:
+      fail-fast: false
+    runs-on: ubuntu-latest
+    steps:
+    - name: Setup Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: ${{ env.GO_VERSION }}
+    - name: Checkout repo
+      uses: actions/checkout@v2
+    - name: Build relic
+      run: make crypto_setup_gopath
+    - name: Docker build
+      run: make install-tools
+    - name: Install Flow Client
+      run: sh -ci "$(curl -fsSL https://storage.googleapis.com/flow-cli/install.sh)"
+    - name: Set up Localnet
+      run: bash -c 'cd integration/localnet/ && make -e OBSERVER=1 init && make start'
+    - name: Ensure Observer is started
+      run: docker ps -q -f name=localnet_observer_1_1
+    - name: Get Client Version ensuring the client is provisioned
+      run: ~/.local/bin/flow -f integration/localnet/flow-localnet.json -n observer version
+    - name: Wait for a default waiting period until a clean state
+      # This will not cause flakiness.
+      # The waiting time is a reasonable time to expect an observer to be responsive
+      run: sleep 10
```

Comment:
Thanks for the review. Well, I gave it a thought after our previous discussion regarding this. Here is the assessment:
1. A waiting period longer than ten seconds already triggers an error. **Common infrastructure should not come up so slow**. Semantically it is valid, but it is not usable. Other tests and customer scenarios will fail.
2. A waiting period less than ten seconds could be achieved by polling up to ten seconds. **Five seconds is a reasonable maximum plus some infrastructure jitter.** However, it is not necessary to poll, since the test takes minutes anyways. Any performance work may address this, if needed.
3. Thus, any flakiness waiting **above ten seconds will show a serious issue in the test infrastructure or the new codebase**, and it should raise an alert.
4. This test says: An observer has a **hard real time requirement** to be responsive in ten seconds on an average test hardware. I think it is reasonable.
5. Adding polling would **signal that this is not production code** to any malware/ransomware, etc., so it should be avoided. Differences reduce use case coverage.

Let me know, if you have any objections.

Reply:
I don't understand what you mean by 5.

Realistically, if this suite becomes flaky, we'll disable it. If you don't want to add the polling that's fine. we'll add it later if/when it becomes a problem.

Reply:
**Summary**

What I mean under non-production code is that it is **different than how the code runs normally**.

Flakiness: In any case, I am happy to add any polling in a later changelist, but I try to piggyback on team member's unit tests that they are writing now, if you do not mind.

**Details**

The goal is maximum coverage. If it does not do something that production does, it is an obvious **loss of coverage**. If it has **additional peculiar tasks** like a polling, it changes the call pattern reducing coverage in a different way.

Many dev tools like fuzzers/optimizers/ai experiments/malware/ransomware/etc. can leverage such a situation. My experience is that such tools add value (except malware/ransomware), but they sometimes add to much cost in terms of additional flakiness, network traffic, debugging time.